scripts/train.py

```diff
@@ -1,55 +1,161 @@
+        train_class_weights = train_class_weights,
+        validation_class_weights = validation_class_weights,
+        learning_rate = float(config['params']['learning_rate']),
+        batch_size = int(config['params']['batch_size'])
+    )
+
+    # Callbacks
+    callbacks = []
+    callbacks.append(LearningRateMonitor(logging_interval="step"))
+    checkpoint_callback = ModelCheckpoint(
+        save_top_k=1,
+        monitor="val_loss",
+        mode="min",
+        dirpath="./checkpoint/",
+    )
+    callbacks.append(checkpoint_callback)
+
+    # configure trainer
+    trainer = pl.Trainer(
+        precision="16-mixed" if cuda_available() else "32-true",
+        callbacks=callbacks,
+        devices= "gpu" if cuda_available() else "auto",
+        accelerator="gpu" if cuda_available() else "auto",
+        log_every_n_steps=1,
+        max_epochs=int(config['params']['max_epochs']),
+    )
+    
+    trainer.fit(net, dm)
+
+    logging.info("Training complete: running test set")
+
+    # Create and log classification report
+    if save_best_model:
+        net_trained = Network.load_from_checkpoint(
+            checkpoint_callback.best_model_path,
+            model_name = config['params']['model_name'],
+            pretrained = config['params']['pretrained'],
+            n_classes = n_classes,
+            dropout = float(config['params']['dropout']),
+            weighted_loss = config['params']['weighted_loss'],
+            train_class_weights = train_class_weights,
+            validation_class_weights = validation_class_weights,
+            learning_rate = float(config['params']['learning_rate']),
+            batch_size = int(config['params']['batch_size'])
+            )
+        trainer.test(
+            net_trained,
+            datamodule=dm,
+        )
+        checkpoint = torch.load(checkpoint_callback.best_model_path)
+
+        file_path = (
+            f"model-{config['project']['name']}-{mlflow.active_run().info.run_name}.pt"
+        )
+
+        checkpoint_info = {}
+        checkpoint_info["monitored_metric"] = checkpoint_callback.monitor
+        checkpoint_info["metric_value"] = checkpoint_callback.best_model_score.item()
+        checkpoint_info["mode"] = checkpoint_callback.mode
+        checkpoint_info["epoch"] = checkpoint["epoch"]
+
+        with open("checkpoint_info.json", "w") as f:
+            json.dump(checkpoint_info, f, indent=2)
+
+        script = torch.jit.script(net_trained._model)
+
+        mlflow.pytorch.log_model(
+            script, file_path, extra_files=["checkpoint_info.json"]
+        )
+
+        _ = [
+            os.remove(fp) for fp in ["checkpoint_info.json"]
+        ]  # remove the temporary files after logging to mlflow
+
 
 if __name__ == '__main__':
 
     if len(sys.argv) > 0:
         config_path = sys.argv[1]
     else:
-        config_path = 'config/config.cfg'
+        config_path = 'config/local_config.cfg'
```

Comment:
Should this be local_config.cfg or config.cfg? I would probably say the latter and let the user adjust it to local_config, but I don't mind if this has thrown people previously.

Again – it would be worth adding a note to this effect in the README. Something like: "default config used by this repo = config.cfg. If you would like to use a local_config file, edit line XX in `train.py` or ensure your config file is called `config.cfg`"

Go with whatever you think is most intuitive/easiest to get started for new people.